subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -65,6 +66,74 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         "project.objects.fileCollection()"   | "project.objects.fileCollection().from(file, symlink, symlinked)"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/1365")
+    def "detect changes to broken symlink outputs"() {
+        def root = file("root").createDir()
+        def target = file("target")
+        def link = root.file("link")
+
+        buildFile << """
+            import java.nio.file.*
+            class ProducesLink extends DefaultTask {
+                @OutputDirectory File outputDirectory // TODO outputfile?
```

Comment:
I think if you add the test for `outputFile` then this PR is good to merge. We should do the improvements (checklist you added to the PR) in a follow up.

Reply:
Great, will do :)

Reply:
Done, please validate if I'm actually reproducing the right thing via the new tests